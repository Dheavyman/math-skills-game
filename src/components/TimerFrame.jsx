import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  minutes: PropTypes.number.isRequired,
  seconds: PropTypes.number.isRequired,
};

/**
 * Timer frame component
 *
 * @param {object} props - Properties passed to component
 *
 * @returns {object} React element
 */
const TimerFrame = (props) => {
  const { minutes, seconds } = props;

  /**
   * Format time
   * Add leading zero when value is less that 10
   *
   * @param {*} value - time to format
   *
   * @returns {object} Formatted time
   */
  const formatTime = value => (
    value < 10 ? `0${value}` : `${value.toString()}`
  );

  return (
    <div className="col-4 col-md-3">
      <h3 className="timer">
        <span className="minutes text-justify">
          {minutes}
          <span>m</span>
        </span>
        <span className="seconds text-justify">
          {formatTime(seconds)}
          <span>s</span>
        </span>
      </h3>
    </div>
  );
};

TimerFrame.propTypes = propTypes;

export default TimerFrame;
