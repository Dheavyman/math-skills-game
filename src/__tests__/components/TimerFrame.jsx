import React from 'react';
import TestRenderer from 'react-test-renderer';

import TimerFrame, { formatTime } from '../../components/TimerFrame';

describe('Timer frame component', () => {
  const props = {
    minutes: 2,
    seconds: 8,
    countdownRunning: false,
    handleChangeTime: jest.fn()
  };

  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<TimerFrame {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});

describe('Format time function', () => {
  it(`should add leading zero to number less that 10 and return the string
    equivalent`, () => {
    const formattedTime = formatTime(5);

    expect(formattedTime).toEqual('05');
  });
  it('should return the string equivalent of numbers greater than 10', () => {
    const formattedTime = formatTime(28);

    expect(formattedTime).toEqual('28');
  });
});
