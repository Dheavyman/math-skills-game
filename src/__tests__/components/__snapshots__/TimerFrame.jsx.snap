// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer frame component should render component successfully 1`] = `
<div
  className="col-4 col-md-3"
>
  <div
    className="timer"
  >
    <h3>
      <span
        className="minutes text-justify"
      >
        2
        <span
          className="minutes-sign"
        >
          m
        </span>
      </span>
      <span
        className="seconds text-justify"
      >
        08
        <span
          className="seconds-sign"
        >
          s
        </span>
      </span>
    </h3>
    <div
      className="btn-group-vertical set-time"
    >
      <button
        className="btn btn-sm"
        name="increase"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-angle-up"
        />
      </button>
      <button
        className="btn btn-sm"
        name="decrease"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-angle-down"
        />
      </button>
    </div>
  </div>
</div>
`;
